src/libmongoc/src/mongoc/mongoc-collection.c

```diff
@@ -2556,13 +2564,14 @@ _mongoc_delete_one_or_many (mongoc_collection_t *collection,
                             bool multi,
                             const bson_t *selector,
                             mongoc_delete_opts_t *delete_opts,
-                            const bson_t *cmd_opts,
-                            bson_t *opts,
+                            const bson_t *extra,
                             bson_t *reply,
                             bson_error_t *error)
 {
    mongoc_write_command_t command;
    mongoc_write_result_t result;
+   bson_t cmd_opts = BSON_INITIALIZER;
+   bson_t opts = BSON_INITIALIZER;
```

Comment:
This replaces the `limit` variable that was previously passed in from the `delete_one` and `delete_many` methods.